src/lib/gravityErrorHandler.ts

```diff
@@ -44,3 +65,37 @@ export const formatGravityError = (error) => {
     return null
   }
 }
+
+export const FieldErrorResultsType = new GraphQLObjectType<
+  any,
+  ResolverContext
+>({
+  name: "FieldErrorResults",
+  fields: () => ({
+    name: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    message: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  }),
+})
+
+type FieldErrorType = {
+  name: string
+  message: string
+}
+
+const formatGravityErrorDetails = (
+  detail: Record<string, string[]>
+): FieldErrorType[] => {
+  const fieldErrors: FieldErrorType[] = []
+
+  Object.keys(detail).forEach((key) => {
+    fieldErrors.push({
+      name: key,
+      message: detail[key].join(", "),
```

Comment:
At first I was wondering why the error about a field would be an array, but I guess there could be multiple errors for a specific field.

The format that the API returns various errors can be found in this file: https://github.com/artsy/gravity/blob/a1ce9f0637d953df0fa10715ed16147c966cc483/app/api/util/error_handlers.rb#L32

The linked line shows the transformation for Mongoid validation errors, and looks like for each field there's an array of messages!

Reply:
Yes these all look to be returned as an array of values.